### PR TITLE
Remove matplotlib agg warning

### LINKED ISSUE #694: Calling "plt.show()" raises an ugly warning
When a script has "plot.show()" in it, it results in the following ugly warning:
<pre>
UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
</pre>
Can we capture it and hide it?

### PR DESCRIPTION
**Pull request by lucyleeow**
closes #694

Did not realise it would be that simple, seems magical.

I didn't add anything to doc assuming no one would miss this warning.

### CODE REVIEW

**Comment by lucyleeow** [comment]
Maybe if anyone raises this we can document how to remove warning filters...

File: sphinx_gallery/tests/test_full.py [hunk 1]
```diff
@@ -748,3 +748,14 @@ def test_minigallery_directive(sphinx_app):
             # Check for examples
             assert explicitorder_example.search(text) is not None
             assert filenamesortkey_example.search(text) is not None
+
```
Comment (GaelVaroquaux) [inline sphinx_gallery/tests/test_full.py:751]: I think that PEP8 wants an extra line return here

File: sphinx_gallery/tests/test_full.py [hunk 2]
```diff
@@ -748,3 +748,14 @@ def test_minigallery_directive(sphinx_app):
...
+def test_matplotlib_warning_filter(sphinx_app):
```
Comment (larsoner) [inline sphinx_gallery/tests/test_full.py:752]: PEP8, two newlines before func def

**Comment by GaelVaroquaux** [comment]
This is green. Merging.

**Comment by lucyleeow** [comment]
Hmm for some reason travis didn't seem to activate, or was that just me?

**Comment by larsoner** [comment]
Looks like the status update just didn't come through, this is green thankfully

https://travis-ci.org/github/sphinx-gallery/sphinx-gallery/builds/691349478

**Pull request merged**